### PR TITLE
feat: edit categories

### PR DESCRIPTION
**Pull request by HarryPotter-4000**
### Description

EXP.CAT-04.1 Edit category

There is a problem and I don't know how to fix it:
When I edited a category from one column, then when opening a category from another, it shows the values ​​of the previously edited category

And we have error react-remove-scroll-bar in console

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Chakra components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [] Console has been checked for warnings

#### Screenshots/Video before change

image

#### Screenshots/Video after change

https://user-images.githubusercontent.com/80061297/184498121-cfc11668-f483-45cc-be55-b49808815eb9.mp4

### CODE REVIEW

**Comment by HarryPotter-4000** [comment]
When I edited a category from one column, then when opening a category from another, it shows the values ​​of the previously edited category

fix that in second commit

File: src/components/AccordionComponent/AccordionComponent.jsx [hunk 1]
```diff
@@ -64,23 +73,24 @@ export const AccordionComponent = (props) => {
...
             variant="danger"
           >
             <DeleteIcon />
           </Button>
 
           <AccordionPanel p={4}>
             <HStack>
-              <Text px={2}>{props.content}</Text>
+              <Text px={2}>{props.name}</Text>
+              {/* ----------Subcategories--------------
```
Comment (kalashnikau) [inline src/components/AccordionComponent/AccordionComponent.jsx:85]: explain please why it's commented?

File: src/components/EditCategoryModal/EditCategoryModal.jsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  useEffect(() => resetForm(), [!isOpen]);
+
+  const closeAllModals = () => {
+    categoriesDeleteModal.onClose();
+    onClose();
+  };
+  //validation does not work, because the form already has default fields, and when compared in isDirty, it considers that everything has been changed
+  const onCancel = () => {
+    isDirty ? categoriesDeleteModal.onOpen() : onClose();
+    // onClose();
```
Comment (kalashnikau) [inline src/components/EditCategoryModal/EditCategoryModal.jsx:75]: why its commented, just remove it

File: src/components/EditCategoryModal/EditCategoryModal.jsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      color: categoryData.color
+    });
+  };
+  useEffect(() => resetForm(), [!isOpen]);
+
+  const closeAllModals = () => {
+    categoriesDeleteModal.onClose();
+    onClose();
+  };
+  //validation does not work, because the form already has default fields, and when compared in isDirty, it considers that everything has been changed
```
Comment (kalashnikau) [inline src/components/EditCategoryModal/EditCategoryModal.jsx:72]: can we do fix for that?

File: src/components/EditCategoryModal/EditCategoryModal.jsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                          'modal.addCategory.validationErrorMessage.pattern'
+                        )
+                      },
+                      validate: (name) =>
+                        (isFetchedCategories &&
+                          !dataCategories.data
+                            .filter(
+                              (data) => data.categoryType === categoryType
+                            )
+                            //.filter((data) => data.id !== categoryData.id)
```
Comment (kalashnikau) [inline src/components/EditCategoryModal/EditCategoryModal.jsx:123]: remove?

**Review by kalashnikau — APPROVED** [review]

File: src/components/EditCategoryModal/EditCategoryModal.jsx (hunk 4 above)
Comment (HarryPotter-4000) [reply src/components/EditCategoryModal/EditCategoryModal.jsx:123]: here I asked the backend to correct their validation and we could change only the color or only the name, I'm waiting for them to do this.

**Review by LizavetaY — DISMISSED** [review]

File: src/components/EditCategoryModal/EditCategoryModal.jsx (hunk 2 above)
Comment (HarryPotter-4000) [reply src/components/EditCategoryModal/EditCategoryModal.jsx:75]: done

File: src/components/AccordionComponent/AccordionComponent.jsx (hunk 1 above)
Comment (HarryPotter-4000) [reply src/components/AccordionComponent/AccordionComponent.jsx:85]: There should be subcategories here, I removed everything that is commented

File: src/components/EditCategoryModal/EditCategoryModal.jsx (hunk 3 above)
Comment (HarryPotter-4000) [reply src/components/EditCategoryModal/EditCategoryModal.jsx:72]: I fix it! and I fix it in AddCategoryModal to!

**Review by LizavetaY — APPROVED** [review]

**Pull request merged**